test/functional/collection_tests.js

```diff
@@ -12,422 +12,405 @@ describe('Collection', function() {
     return setupDatabase(this.configuration);
   });
 
-  /**
-   * @ignore
-   */
-  it('should correctly execute basic collection methods', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
+  describe('', function() {
+    let client;
+    beforeEach(function() {
+      client = this.configuration.newClient(this.configuration.writeConcernMax(), {
```

Comment:
This is a good start for `beforeEach`! Can we move the `client.connect` here also? We can follow the pattern in `crud_spec_tests.js` that looks like this:
```
beforeEach(function() {
    const configuration = this.configuration;
    const client = configuration.newClient();
    return client.connect().then(client => {
      testContext.client = client;
      testContext.db = client.db(configuration.db);
    });
  });
```
I think I might have told you earlier that we don't need to worry about moving `client.connect` and everything else into the `beforeEach`, but after looking at this more I think we should :)

Reply:
Moved client.connect into beforeEach!